src/lib/detection_algorithms/tfrecord/exporter.js

```diff
@@ -0,0 +1,155 @@
+const async = require('async');
+const fs = require('fs');
+const path = require('path');
+const replace = require("replace");
+const detectionUtils = require('../detectionUtils.js');
+const tfrecord = require('tfrecord');
+const CryptoJS = require("crypto-js");
+
+// The Exporter interface - provides a mean to export the tagged frames
+// data in the expected data format of the detection algorithm
+// Constructor parameters:
+//  exportDirPath - path to the directory where the exported file will be placed
+//  classes - list of classes supported by the tagged data
+//  taggedFramesCount - number of positive tagged frames
+//  frameWidth - The width (in pixels) of the image frame
+//  frameHeight - The height (in pixels) of the image frame
+//  testSplit - the percent of tragged frames to reserve for test set defaults to 20%
+function Exporter(exportDirPath, classes, taggedFramesCount, frameWidth, frameHeight, testSplit) {
+    var self = this;
+    self.exportDirPath = exportDirPath;
+    self.labelMap = path.join(self.exportDirPath,"tf_label_map.pbtxt");
+    self.classes = classes;
+    self.taggedFramesCount = taggedFramesCount;
+    self.frameWidth = frameWidth;
+    self.frameHeight = frameHeight;
+    self.testSplit = testSplit || 0.2;
+    self.posFrameIndex = null;
+    self.testFrameIndecies = null;
+
+    // Prepare everything for exporting (e.g. create metadata files,
+    // directories, ..)    
+    // Returns: A Promise object that resolves when the operation completes
+    this.init = function init() {
+         self.posFrameIndex = 0;
+         self.testFrameIndecies = detectionUtils.generateTestIndecies(self.testSplit, taggedFramesCount);
+
+         return new Promise((resolve, reject) => {
+            async.waterfall([
```

Comment:
what is `async.waterfall(...)`.  Is that similar to `Promise.all(...)` to wait for an array of promises to all resolve before resolve the parent promise?

Reply:
It's actually a way to sequentially call a series of functions. This is the pattern the the other vott exporters follow